src/view/xwayland.cpp

```diff
@@ -344,7 +343,11 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
 
         last_server_width = w;
         last_server_height = h;
-        send_configure(w, h);
+
+        if (w != geometry.width && h != geometry.height)
+            send_configure(w, h, XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y |
+                XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT |
+                XCB_CONFIG_WINDOW_BORDER_WIDTH);
```

Comment:
why do we configure border width at all?

Reply:
You're right, this shouldn't be needed.